File: app/src/main/java/com/demeth/massaudioplayer/backend/adapters/SequentialEventManager.java
package com.demeth.massaudioplayer.backend.adapters;

import com.demeth.massaudioplayer.backend.models.adapters.EventManager;
import com.demeth.massaudioplayer.backend.models.objects.Event;

import java.util.LinkedHashMap;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentLinkedQueue;

public class SequentialEventManager implements EventManager {
    private final ConcurrentHashMap<String,EventHandler> handlers;

    public SequentialEventManager(){
        handlers = new ConcurrentHashMap<>();
    }

    @Override
    public void trigger(Event event) {
        handlers.forEach((key, value) -> value.handle(event));
    }

    @Override
    public void registerHandler(String ID, EventHandler handler) {
        handlers.put(ID,handler);
    }

    @Override
    public void removeHandler(String ID) {
        handlers.remove(ID);
    }
}

File: app/src/main/java/com/demeth/massaudioplayer/backend/adapters/SequentialEventManager.kt
package com.demeth.massaudioplayer.backend.adapters

import com.demeth.massaudioplayer.backend.models.adapters.EventHandler
import com.demeth.massaudioplayer.backend.models.adapters.EventManager
import com.demeth.massaudioplayer.backend.models.objects.Event

import java.util.concurrent.ConcurrentHashMap

class SequentialEventManager : EventManager {
    private val handlers: ConcurrentHashMap<String, EventHandler> = ConcurrentHashMap()

    override fun trigger(event: Event) {
        handlers.forEach{ (_, value) ->
            value.handle(event)
        }
    }

    override fun registerHandler(id: String, handler: EventHandler) {
        handlers[id] = handler
    }

    override fun removeHandler(id: String){
        handlers.remove(id)
    }

}
